src/relay.cpp

```diff
@@ -71,6 +71,153 @@ std::map<int, std::string> counterMap = {{DHCPv6_MESSAGE_TYPE_UNKNOWN, "Unknown"
 /* interface to vlan mapping */
 std::unordered_map<std::string, std::string> vlan_map;
 
+
+/* Class Options */
+
+bool Options::Add(OptionCode key, const uint8_t *value, uint16_t len) {
+    std::vector<uint8_t> option_v(value, value + len);
+    options[key] = option_v;
+    return true;
+}
+
+bool Options::Delete(OptionCode key) {
+    if (options.find(key) == options.end()) {
+        return false;
+    }
+    options.erase(key);
+    return true;
+}
+
+std::vector<uint8_t> Options::Get(OptionCode key) {
+    if (options.find(key) == options.end()) {
+        return std::vector<uint8_t>{};
+    }
+    return options[key];
+}
+
+std::vector<uint8_t> *Options::MarshalBinary() {
+    if (options.empty()) {
+        return nullptr;
+    }
+    list.clear();
```

Comment:
Is there a risk of using 'list' recursively?

Reply:
Yes, good point. I will add some protection for this case. And update Class member function descriptions.